atlp-rwanda/eagles-ec-be
# ESLint Configuration PreCommit format and Test

**What does this PR do and Description?**

This feature checks the codes format that you want to commit and converts it into ESLint format which is settled in eslintrc.json file after that it's Tests the all files which is in the Test folder, if all tests pass then it will allows you to commit but if it's fail your code will not be committed

**How to Test this Local ?**

1. Make sure that your codes is ready 
2. and you have pulled the latest changes on your local repository by
_git clone https://github.com/atlp-rwanda/eagles-ec-be_
_cd to eagles-ec-be_
_npm run dev for running the server and connect to DB_
After adding your changes do: 
_git add ._ and _git commit -m "your commit comment"_

Automatically ESLint format will apply itself and test your tests files

**What are the relevant pivotal tracker/Trello stories?**

#187419154

**Some Screenshots For Tested in Local Repository**
The following screenshot below show a successful ESLint Configured and Tests which is PASS[ing]. and Active HoundCI configurations
<img width="1440" alt="Screenshot 2024-04-19 at 15 46 14" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/61088103/59523415-1d25-4cf4-9c5b-96bdb95b9416">
<img width="1440" alt="Screenshot 2024-04-19 at 15 46 32" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/61088103/66dccae4-7061-4da9-9f86-97a764acffbe">
